### PR TITLE
Database manager with inserting user and selecting all from db

### PR DESCRIPTION
**Pull request by andrzejstopka**

### CODE REVIEW

File: backend/db_manager.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                cursor.execute(query, (username, city))
+            self.conn.commit()
+            return json.dumps({"[Message]": "User has been inserted successfully."})
+        except psycopg2.errors.UniqueViolation:
+            self.conn.rollback()
+            return json.dumps({"[Error]": "This user has already provided their location"})
+
+    def get_all(self):
+        cursor = self.conn.cursor(cursor_factory=RealDictCursor)
+        query = "SELECT * FROM user_data"
```
Comment (danielvegas) [inline backend/db_manager.py:35]: It's not a good practice to use *, if you have few fields just type them in.

Comment (andrzejstopka) [reply backend/db_manager.py]: Corrected
2cfaf08b7688624521a1ce8004949f5fe00011ce

File: backend/db_manager.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        cursor = self.conn.cursor()
+        try:
+            if field:
+                query = "INSERT INTO user_data (username, city, field) VALUES (%s, %s, %s)"
+                cursor.execute(query, (username, city, field))
+            else:
+                query = "INSERT INTO user_data (username, city) VALUES (%s, %s)"
+                cursor.execute(query, (username, city))
+            self.conn.commit()
+            return json.dumps({"[Message]": "User has been inserted successfully."})
```
Comment (Hankier) [inline backend/db_manager.py:28]: Not sure what will invoke this function. But maybe its better to create some exit codes  or tuple ( if_success, message. Something that has structure. 
If object of Python will invoke it, how can it tell if function was successful or not? Will you pare a message from it?

Comment (andrzejstopka) [reply backend/db_manager.py:28]: I considered that in backend file will be an object from that function, for example response = DatabaseManager.insert_user() and the response will be sending to Frontend to be displayed. 
So I'm going to think about it and I'll send corrected code.

Comment (andrzejstopka) [reply backend/db_manager.py]: I am not sure that's what you wanted but I have no idea how to do it any other way. 

Could you explain to me why JSON option was wrong? I would like to know for the future.

512cb56b2314da1f6f5bfbd9cfdd63f92f2cd152

Comment (Hankier) [reply backend/db_manager.py]: JSON wasn't bad option.
All I meant that the structure wasn't the same, so it was hard to distinguish if request was successful or not. You might also use sth like this:
```return json.dumps({"success": True, "message": "Sign up successful."})
   return json.dumps({"success": False, "message": "Inserting the user into the database failed."})
   return json.dumps({"success": False, "message": "User already exists."})  
```

Fields in json structure are the same and you can check if query was succesful. :)

Comment (andrzejstopka) [reply backend/db_manager.py]: I have already understood, so I turned it back to JSON, thanks :)

**Review by mjzalewska — APPROVED** [review]

File: backend/db_manager.py [hunk 3]
```diff
@@ -25,10 +25,12 @@ def insert_user(self, username, city, *field):
                 query = "INSERT INTO user_data (username, city) VALUES (%s, %s)"
                 cursor.execute(query, (username, city))
             self.conn.commit()
-            return (True, "User has been inserted successfully.")
+            return json.dumps({"success": True, "message": "Sign up successful."})
         except psycopg2.errors.UniqueViolation:
             self.conn.rollback()
-            return (False, "This user has already provided their location.")
+            return json.dumps({"success": False, "message": "User already exists."})
+        except:
```
Comment (bartoszkobylinski) [inline backend/db_manager.py:32]: Why there is no Exception added?
It should always be general Exception handled while you try to catch some particular Exception.
If you are returning error messages using your customize solution like here is a json file it's wort to add why specificly you got error message

**Pull request merged**